node/actors/supervisor.go

```diff
@@ -18,27 +29,55 @@ type NexSupervisor struct {
 func (sup *NexSupervisor) Init(args ...any) (act.SupervisorSpec, error) {
 	var spec act.SupervisorSpec
 
-	nodeOptions := args[0].(models.NodeOptions)
+	if len(args) != 3 {
```

Comment:
see @bruth 's previous comment. Things get a lot cleaner if we pass a single struct that holds the node id, the connection, and the node options.